### PR TITLE
feat(testing): add Effect Vitest canon foundation

### PR DESCRIPTION
**Pull request by kriegcloud**
This adds the Effect Vitest canon foundation: a syntax-only idiom detector and
keyed baseline, a source-pinned primitive graph, migration charters, and an
instrumented Effect test runner. The runner records a separate lifecycle for each
test and diagnoses stalled work with a live-clock watchdog.

Main's Effect and @effect/vitest 4.0.0-rc.113 upgrade is integrated. Native
property callbacks retain their lifecycle context across Effect resumptions.
The shared FileSystem conformance helper includes rc.113's new cases and stronger
read, truncate and write checks, preserving per-test resource ownership.

The final review fixes detect scoped function references and curried provider
aliases at their application sites. Whole-body scopes are distinguished from
shorter lifetimes; lexical provenance, shadowing, pure stubs and matcher operands
remain intact. Outcome hints now use neutral guidance when the asserted branch
is unknown. Traversal avoids redundant lookups and empty token-child arrays while
preserving every complete finding record.

Local validation for the final source:

- Full CLI package audit and docgen pass in 420.944 seconds. All 288 focused
  detector/Knowledge cases pass, with prior assertions retained apart from the
  reviewed correction to the EV005 expected hint.
- Normal scoped Node 22 coverage passes 3,656 tests across 181 files, with five
  existing skips. Knowledge branch coverage is 95.60% against a 95.31% floor;
  detector function coverage is 98.12% against 97.63%. Baseline bytes are unchanged.
- All 52 runner registrations retain their expected outcomes on Node 22, Node 24
  and Bun. Full test-utils package verification and scoped Node 22 coverage pass.
- The adopted inventory contains 8,138 open candidates across 1,000 tests and
  110 support modules. Every transition from 7,775 rows is reconciled: 7,748 exact
  survivors, 27 approved outcome-hint changes, 363 additions and zero removals or
  exception transfers. The final performance changes preserve all 8,138 payloads.
- Three ordinary commands pass in 9.563, 9.630 and 9.926 seconds. Raw host pressure,
  memory, swap and CPU context and all failed cohorts remain recorded. A separate
  concurrent cheap-gate invocation took 10.025 seconds; these are measured local
  results with limited margin, not load-normalized or worst-case guarantees.
- All three adversarial rounds are closed. On `a8a2923ca6`, all 18 required
  hosted checks pass and 17 optional checks pass or are skipped. Greptile reports
  5/5 with zero issues and zero unresolved review threads; Yeet reports
  `merge-ready: yes`.
- The full local Node 24 run failed an inherited ACP JSON round-trip property.
  The unchanged test reproduces that escaped-key failure on Node 24.20.0 and
  passes with the same seed on Node 22.22.3, which hosted coverage already pins
  for this runtime bug. No test, assertion or coverage baseline was changed.
- A full Yeet rerun with Node 22 passed preflight, all 140 build tasks, 1,893
  doctests, full docgen, integration, lint/policy, typechecking, all 135 unit-test
  tasks and the completed Storybook variants. It was intentionally interrupted
  after GitHub reported this PR merged as `955528adb4`. This is not counted as
  a completed aggregate pass; the completed scoped proofs and all hosted checks
  above remain the final validation evidence.

The installed stack retains main's Vitest 4.1.11. Exercised behavior passes while
rc.113 declares a Vitest 5 peer range. No property floors, coverage floors or global
timeout policy are weakened. The nonempty detector baseline is intentional for
this foundation PR. P1 inventory and P2 migration retain their separate packet
gates, and Benjamin retains plan ratification and merge approval.

Evidence: `goals/effect-vitest-canon/history/2026-09-10-adversarial-round-3-closure.md`
and `goals/effect-vitest-canon/history/2026-09-10-r3-performance.md`.

---
<a href="https://app.blacksmith.sh/beep-effect/codesmith/beep-effect/pr/1067"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1791570175&installation_model_id=424656&pr_number=1067&repository=beep-effect%2Fbeep-effect&return_to=https%3A%2F%2Fgithub.com%2Fbeep-effect%2Fbeep-effect%2Fpull%2F1067&signature=4a2c2b565c3c78be3c99c1a32f211ddf0f6196bbab1cc8e6ad5261f53f01d9c7"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

---

## Provenance

- Workspace: <code>effect-vitest-canon</code>
- Branch: <code>feat/effect-vitest-canon</code>
- Agents (newest first):
  - `codex` (tui) · `unknown` · monitored
  - `unknown` (unknown) · `unknown` · monitored

Resume the publishing agent from any beep-effect checkout on the publishing workstation:

```sh
bun run beep yeet resume 1067
```

### CODE REVIEW

File: packages/tooling/tool/cli/src/commands/Lint/internal/EffectVitestScan.ts [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+  const census = A.empty<EffectVitestCensusRow>();
+  const findings = A.empty<EffectVitestFinding>();
+  const encoder = new TextEncoder();
+
+  const project = new Project({ skipAddingFilesFromTsConfig: true, skipFileDependencyResolution: true });
+  project.addSourceFilesAtPaths(sourcePaths);
+  yield* Console.log(`[effect-vitest:phase] projectMs=${(performance.now() - started).toFixed(1)}`);
+  for (const sourceFile of project.getSourceFiles()) {
+    const absolute = sourceFile.getFilePath();
+    const file = Str.replace(`${root}/`, "")(absolute);
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/tooling/tool/cli/src/commands/Lint/internal/EffectVitestScan.ts]: Fixed in aa8143e7f79d4202c1952536cbb74dda684b763e. The scanner now derives the repository-relative path with Path.relative and normalizes it with the shared toPosixPath helper. Drive-letter and UNC regressions compare the complete emitted finding payload and ratchet identity, not just a path substring. All 111 focused scanner tests and full repo-cli package audit/docgen pass; the regenerated canonical inventory and ordinary ratchet also pass.

File: packages/tooling/test-kit/test-utils/src/test/Vitest.test-kit.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * **Example** (Bind a watchdog without replacing the body clock)
+ *
+ * ```ts
+ * import { makeIt } from "@beep/test-utils/test/Vitest";
+ * import { expect } from "@effect/vitest";
+ * import * as Clock from "effect/Clock";
+ * import * as Effect from "effect/Effect";
+ * import { TestClock } from "effect/testing";
+ *
+ * const it = makeIt(Effect.runSync(Clock.Clock));
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply packages/tooling/test-kit/test-utils/src/test/Vitest.test-kit.ts:32]: Verified against the pinned rc.113 API: [Clock.Clock is a Context.Reference<Clock>](https://github.com/Effect-TS/effect/blob/d3b837aee836f35d625d55205f7d6e61305fc198/packages/effect/src/Clock.ts#L189), and [Context.Key extends Effect](https://github.com/Effect-TS/effect/blob/d3b837aee836f35d625d55205f7d6e61305fc198/packages/effect/src/Context.ts#L64). Reference supplies a default service with no required environment. The exact unchanged example compiles with the installed tsgo in ESM context; full test-utils package audit/docgen passes. Node 24.20.0 and Bun 1.4.2 also resolve Effect.runSync(Clock.Clock) successfully and read its current/monotonic time effects. The example in aa8143e7f79d4202c1952536cbb74dda684b763e is valid, so no source correction is needed for this finding.
